iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/hsm/IotHubSasTokenHsmAuthenticationProvider.java

```diff
@@ -27,9 +27,8 @@ public class IotHubSasTokenHsmAuthenticationProvider extends IotHubSasTokenWithR
      * @param signatureProvider the signature provider to be used when generating sas tokens
      * @param deviceId the id of the device the module belongs to
      * @param moduleId the id of the module to be authenticated for
-     * @param hostname the hostname of the iothub to be authenticated for. May be null if either gatewayHostname or mqttGatewayHostname is not
-     * @param gatewayHostname the gatewayHostname of the edge hub to be authenticated for. May be null if either hostname or mqttGatewayHostname is not
-     * @param mqttGatewayHostname the mqttGatewayHostname of the edge hub to be authenticated for. May be null if either hostname or gatewayHostname is not
+     * @param hostname the hostname of the iothub to be authenticated for. May be null if gatewayHostname is not
```

Comment:
I didn't know that hostname could be null. Wouldn't the gateway still need to know which upstream hub to route to?

Reply:
This is reverted to what it originally was. But with another look, this statement that hostname could be null is strange indeed as we have some null checks regarding hostName in the downstream logics, like [here](https://github.com/Azure/azure-iot-sdk-java/blob/previews/E4Kadapter/iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubAuthenticationProvider.java#L32). 

@timtay-microsoft Do you have any idea why we have this comment or any objection of removing it?

Reply:
Allowing the hostname to be null seems wrong to me as well. Go ahead and make it mandatory